### PR TITLE
fix the detection of UUID columns for tables in avro format

### PR DESCRIPTION
**Pull request by Anna050689**
fix the detection of UUID columns

### CODE REVIEW

File: src/syngen/ml/vae/models/dataset.py [hunk 1]
```diff
@@ -417,13 +420,12 @@ def _avro_data_pipeline(self, df, schema):
...
         self.str_columns = set(column for column, data_type in schema.items() if data_type == 'string')
         self.categ_columns -= self.long_text_columns
-        self.str_columns = \
-            self.str_columns - self.categ_columns - self.int_columns - self.binary_columns - self.long_text_columns
-        self.date_columns = \
-            get_date_columns(df, list(self.str_columns)) - self.categ_columns - \
-            self.binary_columns - self.long_text_columns
+        self.str_columns = self.str_columns - self.categ_columns - self.binary_columns \
+                           - self.long_text_columns - self.uuid_columns
+        self._set_date_columns(df)
```
Comment (pbobyrev) [inline src/syngen/ml/vae/models/dataset.py:425]: What's the purpose of the second function execution here? The first one was on line 402 in the new code.

File: src/tests/unit/dataset/fixtures/table_with_diff_uuid_columns_with_missing_values.csv [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+20caf14c-1a43-1b5d-b5a9-338905804ec6,5e493a5a-a585-2d42-9d87-582cc8ab841a,,9ee16ad1-c70b-4aae-953b-b9b9acec4cb0,,01H1S84KY33A54TC5XRCJNZMZT
+71bf6bcf-4d48-16de-8bbb-9ee74285acec,,0e89d99e-4c4e-398c-976c-146fa28fc1a3,3c26db07-4c19-4556-b3fc-b24426c8076e,,01H1S84KY340DXDX4CKM92TZ20
+,c24ab909-af55-2c08-8737-3533b2c4e950,,,d40cf41c-b2fa-58c3-a41a-dec8af34dccb,
+0488d169-30e4-14cb-b1a8-6288e63bb18e,d4bec70d-ac42-2488-8242-9f646bf9a2f4,42f332ed-1fc9-35de-8cf1-814175d47201,,a7076855-11b9-5993-a3c1-d627b9ae43b7,01H1S84KY3CNNZHGXW9VM47W47
+,,9bd9371a-35ac-3a58-b739-fafc8a871f1b,b27e7c91-2d69-4494-935e-3552aacb2e12,5c26b5dd-e4dc-5bb0-b2d7-3e6eee137bd4,
+,,036941cf-75d6-36da-a752-8778d1f16fde,b74b3577-8435-42b7-9459-daed65f58447,,
+281bdf03-33e9-18e4-a55e-8cd02a960ca8,bf01e54c-b06a-2c16-a642-dd0720db6713,,e0e02f3d-4620-4298-a3c0-653dc5511477,7376c1c9-7687-517d-bbf8-d8b968100715,01H1S84KY3HHGHXCWB10M0N6BH
+ed0c1449-19e7-1ff3-b8fd-90cb493ac1a3,0ef3cbde-506b-2c3c-93dd-f3b79f0480d1,938b6bbf-a4e8-3002-b094-6a1fc626f832,,37948870-1b2f-580f-be4c-da904290c32e,01H1S84KY369THXP0DS3EP4RJE
```
Comment (pbobyrev) [inline src/tests/unit/dataset/fixtures/table_with_diff_uuid_columns_with_missing_values.csv:101]: Thank you for writing the test!

**Review by pbobyrev — COMMENTED** [review]
Ann, there is a comment in a place I am not fully understand what's going on

File: src/syngen/ml/vae/models/dataset.py (hunk 1 above)
Comment (Anna050689) [reply src/syngen/ml/vae/models/dataset.py:425]: There are 2 different pipelines - _general_data_pipeline, _avro_data_pipeline

**Pull request merged**